src/dafne/utils/sam_mask_refine.py

```diff
@@ -146,93 +201,121 @@ def generate_points_from_mask(mask):
     return np.array(point_list)
 
 
+def determine_device():
+    if GlobalConfig['USE_GPU_FOR'] != 'Autosegmentation' and torch.cuda.is_available():
+        print('SAM loaded on GPU')
+        return 'cuda'
+    print('SAM loaded on CPU')
+    return 'cpu'
+
+
 def enhance_mask(img, mask, progress_callback: Optional[Callable[[int, int], None]] = None):
-    global predictor, old_img
+    global old_img, image_embedding, predictor
 
     # if there is no mask, return the original mask
     if not mask.any():
         return mask
 
-    if GlobalConfig['SAM_MODEL'] == 'Large':
-        model_type = 'vit_h'
-    elif GlobalConfig['SAM_MODEL'] == 'Medium':
-        model_type = 'vit_l'
-    elif GlobalConfig['SAM_MODEL'] == 'Small':
-        model_type = 'vit_b'
+    device = determine_device()
+    model_choice = GlobalConfig['SAM_MODEL']
 
     def show_progress(current, maximum):
         if progress_callback is not None:
             progress_callback(current, maximum)
 
     show_progress(0, 100)
 
-    if predictor is None:
-        print('Loading SAM...')
-        predictor = load_sam(model_type, progress_callback)
+    print('Loading SAM...')
+    sam = load_sam(model_choice, progress_callback)
 
     show_progress(30, 100)
 
     if img is not old_img:
         print('Loading image...')
         old_img = img
         img_norm = img * 255 / img.max()
-        predictor.set_image(np.stack([img_norm, img_norm, img_norm], 2).astype(np.uint8))
+        image_embedding = None
+        predictor = None
 
-    show_progress(80, 100)
-
-    point_list = generate_points_from_mask(mask)
     bbox = enlarge_bounding_box(mask, GlobalConfig['SAM_BBOX_EXPAND_FACTOR'])
 
-    # generate negative points
-    negative_mask = np.logical_not(mask)
-    bbox_region = np.zeros_like(mask)
-    bbox_region[bbox[1]:bbox[3], bbox[0]:bbox[2]] = 1
-    negative_mask = np.logical_and(negative_mask, bbox_region)
-
-    negative_point_list = generate_points_from_mask(negative_mask)
-
-    t = time.perf_counter()
+    if model_choice in ['Med Sam']:
+        H, W = img.shape[:2]
+        if image_embedding is None:
+            img_3c = np.repeat(img_norm[:, :, None], 3, axis=-1) if len(img_norm.shape) == 2 else img_norm
+            img_1024 = transform.resize(img_3c, (1024, 1024), order=3, preserve_range=True, anti_aliasing=True).astype(np.uint8)
```

Comment:
I'm not sure if that's the line causing the issue, since it's used in MedSAM example as well - [example MedSAM Colab link](https://colab.research.google.com/drive/19WNtRMbpsxeqimBlmJwtd1dzpaIvK2FZ?usp=sharing#scrollTo=dXvYXPRwufXi)

do you have any ideas on how to fix this?
maybe `img_1024 = cv2.resize(img_3c, (1024, 1024), interpolation=cv2.INTER_CUBIC)`?